### DISCUSSION TITLE
How to debug a .vsix plugin

### DISCUSSION THREAD

**Discussion by nmaaiken** [eclipse-theia/theia#10053] [Q&A] [answered]
I have a custom filesystem extension in .vsix format.
It works fine with VS Code but when using it in Theia, some errors occur. How can i debug the extension within Theia?
I can't find any plugins in the browser's debugger and searching for keywords yields nothing as well.

**Comment by msujew** [comment, marked as answer]
Hi @nmaaiken, thank you for your question.

You can debug the extension by first starting the Theia backend in debug mode and after connecting a frontend to the application, attach to the plugin host process with the following launch config. Afterwards you can debug the vscode extension. The code that's responsible for running the extension is [here](https://github.com/eclipse-theia/theia/tree/master/packages/plugin-ext/src/plugin). Keep in mind, that all extension code is running in the backend, so you'll have to debug it with vscode/theia instead of using the browser's debugger.

https://github.com/eclipse-theia/theia/blob/d3501165bb4e87c3612a1a02c34a1d16ab81802c/.vscode/launch.json#L81-L85
